exporter/exporterhelper/queued_retry_inmemory.go

```diff
@@ -43,6 +40,12 @@ type QueueSettings struct {
 	NumConsumers int `mapstructure:"num_consumers"`
 	// QueueSize is the maximum number of batches allowed in queue at a given time.
 	QueueSize int `mapstructure:"queue_size"`
+	// StorageID if not empty, enables the persistent storage and uses the component specified
+	// as a storage extension for the persistent queue
+	StorageID *config.ComponentID `mapstructure:"storage"`
+	// StorageEnabled describes whether persistence via a file storage extension is enabled using the single
+	// default storage extension.
+	StorageEnabled bool `mapstructure:"persistent_storage_enabled"`
```

Comment:
I don't like this config, is too complex for no good reason. I see no point in supporting `persistent_storage_enabled` instead of simply asking for the exact name.

Reply:
The current state was an attempt to keep the config backwards-compatible, as this whole change was originally just about enabling the queue in the build by default. If we're going to break it, then I agree with @bogdandrutu - just make the user specify the storage explicitly and be done with it, looks like the best option in the long term.

If we do that though, I think the changelog should mention the change explicitly.

Reply:
I would not call this a breaking change since this was an experimental feature.

Reply:
It's not technically a breaking change, but it's nonetheless a feature being used in the real world, and I think it'd be polite to call it out in the changelog to make it easier for users to figure out what broke their config after an upgrade.

Either way, I'll make the change in this PR.

Reply:
FYI: Waiting for the change.

Reply:
Done, sorry for the delay. I've added a changelog entry to breaking changes, even though this technically isn't one.